beacon-chain/sync/validators_custody.go

```diff
@@ -0,0 +1,147 @@
+package sync
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/prysmaticlabs/prysm/v5/async"
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/core/feed"
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/core/feed/state"
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/core/peerdas"
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/p2p"
+	"github.com/prysmaticlabs/prysm/v5/config/params"
+	"github.com/prysmaticlabs/prysm/v5/consensus-types/primitives"
+	"github.com/prysmaticlabs/prysm/v5/time/slots"
+)
+
+const updateToAdvertiseCustodyGroupCountPeriod = 1 * time.Minute
+
+func (s *Service) maintainValidatorsCustody() {
```

Comment:
instead of running it on every block, could you have it be triggered every minute/epoch and then just use the head state ?

Reply:
Fixed in https://github.com/prysmaticlabs/prysm/pull/14948/commits/5c03353d8570848b7dce99469eff95fb72523bd4.